app/src/main/java/com/github/sdp/ratemyepfl/database/reviewable/CourseRepositoryImpl.kt

```diff
@@ -0,0 +1,101 @@
+package com.github.sdp.ratemyepfl.database.reviewable
+
+import com.github.sdp.ratemyepfl.database.DatabaseException
+import com.github.sdp.ratemyepfl.database.Repository
+import com.github.sdp.ratemyepfl.database.query.Query.Companion.DEFAULT_QUERY_LIMIT
+import com.github.sdp.ratemyepfl.database.query.QueryResult
+import com.github.sdp.ratemyepfl.database.query.QueryResult.Companion.asQueryResult
+import com.github.sdp.ratemyepfl.database.reviewable.ReviewableRepositoryImpl.Companion.AVERAGE_GRADE_FIELD_NAME
+import com.github.sdp.ratemyepfl.database.reviewable.ReviewableRepositoryImpl.Companion.NUM_REVIEWS_FIELD_NAME
+import com.github.sdp.ratemyepfl.model.items.Course
+import com.github.sdp.ratemyepfl.model.review.ReviewRating
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.ktx.getField
+import kotlinx.coroutines.flow.zip
+import kotlinx.coroutines.tasks.await
+import javax.inject.Inject
+
+class CourseRepositoryImpl private constructor(private val repository: ReviewableRepositoryImpl<Course>) :
+    CourseRepository,
+    ReviewableRepository<Course> by repository, Repository<Course> by repository {
+
+    @Inject
+    constructor(db: FirebaseFirestore) : this(
+        ReviewableRepositoryImpl(
+            db,
+            COURSE_COLLECTION_PATH,
+            COURSE_CODE_FIELD_NAME,
+        ) { documentSnapshot ->
+            documentSnapshot.toCourse()
+        })
+
+    companion object {
+        const val COURSE_CODE_FIELD_NAME: String = "course_code"
```

Comment:
Should be 'courseCode', it's like this on the DB.

Reply:
Yes, changed it when I add it in the db (to courseCode)